pkg/flist.go

```diff
@@ -28,6 +30,17 @@ type MountOptions struct {
 	Type DeviceType
 }
 
+// Valid checks that mount options are valid
+func (m *MountOptions) Valid() error {
+	// we disable HDD support in flist creation. root fs of containers
+	// should always be SSD
+	if m.Type != SSDDevice {
```

Comment:
I don't think this belong at this level. Flistd should keep this capability. The reason is we want to be able to still mount flist using any type of disk for internal usage (upgrade, ...).

This check should be at provisiond instead.